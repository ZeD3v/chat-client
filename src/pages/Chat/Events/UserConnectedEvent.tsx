import React from 'react';
import { format } from 'date-fns';

import { User } from '../../../App';
import { ChatEvents } from '../index';

import styles from './styles/ConnectionStatusEvent.module.scss';

type UserConnectedEventPayload = {
  username: string;
};

type UserConnectedEvent = {
  type: ChatEvents.USER_CONNECTED;
  id: string;
  timestamp: number;
  payload: UserConnectedEventPayload;
};

type UserConnectedEventProps = {
  user: User | null;
  event: UserConnectedEvent;
};

const UserConnectedEvent: React.FC<UserConnectedEventProps> = ({
  user,
  event,
}: UserConnectedEventProps) => (
  <div className={styles.container}>
    {user?.username === event.payload.username ? 'You' : event.payload.username}{' '}
    joined - {format(new Date(event.timestamp), 'dd/MM/yyyy HH:mm:ss')}
  </div>
);

export default UserConnectedEvent;
