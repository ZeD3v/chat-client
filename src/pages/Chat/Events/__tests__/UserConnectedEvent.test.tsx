import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import UserConnectedEvent from '../UserConnectedEvent';

import { ChatEvents } from '../../index';

describe('UserConnectedEvent', () => {
  test('renders correctly for the user', () => {
    const { asFragment } = render(
      <UserConnectedEvent
        user={{ username: 'Test user 1', token: 'test-token' }}
        event={{
          id: 'testid-1',
          timestamp: 1589041780,
          type: ChatEvents.USER_CONNECTED,
          payload: {
            username: 'Test user 1',
          },
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly for other users', () => {
    const { asFragment } = render(
      <UserConnectedEvent
        user={{ username: 'Test user 1', token: 'test-token' }}
        event={{
          id: 'testid-1',
          timestamp: 1589041780,
          type: ChatEvents.USER_CONNECTED,
          payload: {
            username: 'Test user 2',
          },
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
