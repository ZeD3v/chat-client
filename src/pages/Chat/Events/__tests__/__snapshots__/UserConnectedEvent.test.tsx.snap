// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserConnectedEvent renders correctly for other users 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    Test user 2 joined - 19/01/1970 12:24:01
  </div>
</DocumentFragment>
`;

exports[`UserConnectedEvent renders correctly for the user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    You joined - 19/01/1970 12:24:01
  </div>
</DocumentFragment>
`;
